src/basic/unit-file.c

```diff
@@ -412,12 +413,25 @@ int unit_file_build_name_map(
                 if (null_or_empty_path(dst) != 0)
                         continue;
 
-                /* Do not treat instance symlinks that point to the template as aliases */
-                if (unit_name_is_valid(basename(dst), UNIT_NAME_TEMPLATE) &&
-                    unit_name_is_valid(src, UNIT_NAME_INSTANCE))
-                        continue;
+                dst = basename(dst);
```

Comment:
I think we prefer the `path_extract_filename()` helper to `basename` these days

Reply:
The names that are used here are very tightly controlled, and have been through verification already. So the additional checks that path_extract_filename() does are not useful. And basename() does a very simple thing here and we don't need anything more fancy.

Also, if this were to change, then the whole file should be updated, it would be confusing to have `path_extract_filename()` here just because the code was moved around a bit, but `basename()` in other places.

Reply:
Ok, makes sense

Reply:
in the long run we really should move everything over though. you never know what basename() returns...